src/condor_q.V6/jobmatch.cpp

```diff
@@ -849,61 +910,121 @@ bool doJobRunAnalysis (
 			if ( ! offer->LookupBool("DynamicSlot", is_dslot)) is_dslot = false;
 		}
 
-		// ignore dslots
-		if (is_dslot && (mode == anaMatchModePslot))
+		// update counts by group, where in this case slots are grouped by STARTD
+		auto & groupCounts = startdAds.groupIdCount[slotad.group_id];
+		if ( ! groupCounts.tot) {
+			// this is the first time i have seen this group/machine
+			ac.totalUniqueMachines++;
+		}
+		++groupCounts.tot;
+		if (is_pslot) groupCounts.major++;
+
+		// ignore dslots if the match mode is Pslot (and static) only
+		if (is_dslot && (mode == anaMatchModePslot)) {
+			ac.totalDSlotsSkipped++;
 			continue;
+		}
 
-		slotname = "undefined";
-		offer->LookupString(ATTR_NAME, slotname);
+		auto & machine = slotad;
+		groupCounts.considered = 1;
 
+		bool within_limits = true; // within limits of original slot ad
 		autoOverlayAd overlay; // in case we need a pslot overlay ad, this will automatically delete it when it goes out of scope
 		if (is_pslot && (mode >= anaMatchModePslot)) {
 			ExprTree * expr = offer->Lookup(ATTR_WITHIN_RESOURCE_LIMITS);
 			classad::Value val;
-			bool within = false;
-			if (expr && EvalExprToBool(expr, offer, request, val) && val.IsBooleanValueEquiv(within) && ! within) {
+			if (expr && EvalExprToBool(expr, offer, request, val) && val.IsBooleanValueEquiv(within_limits) && ! within_limits) {
 				// if the slot doesn't fit within the pslot, try applying an overlay ad that restores the pslot to full health
 				ClassAd * pov = make_pslot_overlay_ad(offer);
 				if (pov) {
-					overlay.chain(pov, offer);
-					offer = pov; // match using the pslot overlay
+					std::string adbuf, slotname;
+					offer->LookupString(ATTR_NAME, slotname);
+					dprintf(D_ZKM, "creating drain simulation overlay for pslot %s :\n%s",
+						slotname.c_str(), formatAd(adbuf, *pov, "\t", nullptr, false));
```

Comment:
I thought `D_ZKM` wasn't supposed to appear in production code

Reply:
this is in a tool,  so not really a problem.